yt_dlp/utils/networking.py

```diff
@@ -79,6 +79,22 @@ def __contains__(self, key):
         return super().__contains__(key.title() if isinstance(key, str) else key)
 
 
+class CaseSensitiveDict(HTTPHeaderDict):
```

Comment:
I think we should put this into the extractor file, since its only a temporary workaround until we make `HTTPHeaderDict` store sensitive keys and the handler extension will select pre-normalization of headers

Reply:
My plan was to have the proper solution merged first in core, and then rebase this PR.
The current solution is only committed to allow for testing.

If writing a proper solution turns out more difficult, we can move `CaseSensitiveDict` in to the extractor to get this PR merged.